### PR TITLE
Active repos user has permission to endpoint

### PR DESCRIPTION
**Pull request by oudeismetis**
Connected to #35

### CODE REVIEW

File: franklin/config/settings/base.py [hunk 1]
```diff
@@ -106,8 +106,8 @@
     'social.pipeline.user.get_username',
     'social.pipeline.user.create_user',
     'social.pipeline.social_auth.associate_user',
-    'core.views.save_oauth',
     'social.pipeline.social_auth.load_extra_data',
+    'core.views.save_oauth',
```
Comment (oudeismetis) [inline franklin/config/settings/base.py]: moving this 1 step later fixes a bug with our current social auth implementation

File: franklin/builder/serializers.py [hunk 2]
```diff
@@ -14,13 +14,26 @@ class Meta:
...
+        extra_kwargs = {
+            "github_id": {
+                "validators": [],
+            },
+        }
+
 
 class SiteSerializer(serializers.ModelSerializer):
     owner = OwnerSerializer()
+    environments = EnvironmentSerializer(many=True, required=False)
```
Comment (oudeismetis) [inline franklin/builder/serializers.py]: Adding this and setting it to `require=False` Lets us use this serializer is 2 ways:
- When registering a new project for the first time, we don't need/want environment details and just use the site and owner details to initialize everything.
- When the dashboard wishes to list details about active projects, we want it to also fetch details about the environments (url, status, etc.)

File: franklin/github/views.py [hunk 3]
```diff
@@ -89,11 +89,11 @@ def register_repo(request):
     responseMessages:
         - code: 400
           message: Invalid json received or Bad Request from Github
+        - code: 403
+          message: Current user does not have deployment permissions
         - code: 422
           message: Validation error from Github
     """
```
Comment (oudeismetis) [inline franklin/github/views.py]: The registration endpoint now requires admin level permissions for a repo. We don't "need" this yet, but we will when we generate a deploy key. Builder will need that to properly clone repos and only admins of a repo can add new deploy keys.

Keep that in mind if you are testing API and are getting an error for a test project that used to work.

**Comment by CorneliusIV** [comment]
:+1:

**Pull request merged**